File: src/learning_api/src/spin_function.cpp
#include <ros/ros.h>
#include "std_msgs/String.h"

void Callback(const std_msgs::String::ConstPtr& msg)
{
    ROS_INFO("AAA");
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "spin_function_cpp");

    ros::NodeHandle n;

    ros::Subscriber subscriber = n.subscribe("/initialization", 10, Callback);

    ros::spin();
    //ros::spinOnce();

    ROS_INFO("BBB");

    return 0;
}

File: src/learning_api/scripts/spin_function.py
#!/usr/bin/env python3

import rospy
from std_msgs.msg import String

def Callback(msg):
    rospy.loginfo("AAA")

if __name__ == '__main__':
    
    rospy.init_node('learning_api_py', anonymous = True)
    
    subscriber = rospy.Subscriber('/initialization', String, Callback)

    rospy.spin()

    rospy.loginfo("BBB")